zadanie 2

exmortuo opened the pull request.

exmortuo committed:
zadanie 2

BernardBoguszewski commented on src/main/java/pl/sdacademy/javalub11/exercises/Greetings.java:
```diff
@@ -3,7 +3,11 @@
 
     public String greet(String name) {
-        return "Witaj, " + name;
+        if(name != null) {
```
format kodu - ctrl + alt + L

BernardBoguszewski commented on src/test/java/pl/sdacademy/javalub11/exercises/GreetingsTest.java:
```diff
@@ -20,5 +20,15 @@ public void testShouldReturnProperGreeting() {
     }
 
+    @Test
+    public void testParameterGreetingsMethodShouldBeNotNull(){
```
testShould na początku nazwy metod testujących

exmortuo committed:
Zad 1, wymaganie 3

BernardBoguszewski commented on src/test/java/pl/sdacademy/javalub11/exercises/GreetingsTest.java:
```diff
@@ -31,4 +31,17 @@ public void testParameterGreetingsMethodShouldBeNotNull(){
+        Greetings greetings = new Greetings();
+
+        //when
+        String name = "ADAM";
```
zmienna name w sekcji given

BernardBoguszewski commented on src/test/java/pl/sdacademy/javalub11/exercises/GreetingsTest.java:
```diff
@@ -31,4 +31,17 @@ public void testParameterGreetingsMethodShouldBeNotNull(){
+        //when
+        String name = "ADAM";
+        String result = greetings.greet(name);
+
```
spacje

exmortuo committed:
Zad 1, wymaganie 4

and added changes to response on Your comments in
Zad. 1, wymagania 3

BernardBoguszewski commented on src/main/java/pl/sdacademy/javalub11/exercises/Greetings.java:
```diff
@@ -4,12 +4,21 @@
                 return greetings.toUpperCase();
             } else {
-                return "Witaj, " + name;
+                String[] names = name.split(",");
```
można wydzielić nową metodę prywatną

exmortuo committed:
Zad 1, wymaganie 5

and code where is generate greetings string for names from list
was moved to other, private method -> zad. 1 wymagania 4

exmortuo committed:
Zad 1, wymaganie 6

exmortuo committed:
Zad 1, wymaganie 7

exmortuo committed:
Zad 2 - Square Brackets

exmortuo committed:
Zad 3. wymaganie 1

exmortuo committed:
Zad 3. wymaganie 2

exmortuo committed:
Zad 3. wymaganie 3

exmortuo committed:
Zad 3. wymaganie 4

exmortuo committed:
Zad 3. wymaganie 5

BernardBoguszewski commented on src/main/java/pl/sdacademy/javalub11/exercises/SquareBrackets.java:
```diff
@@ -0,0 +1,36 @@
+
+        int balanceBetweenBrackets = 0;
+
+        for (int i = 0; i < bracketsArray.length; i++) {
```
można wyciągnąć strategie rozwiązania do metod prywatnych

BernardBoguszewski commented on src/test/java/pl/sdacademy/javalub11/exercises/SquareBracketsTest.java:
```diff
@@ -0,0 +1,21 @@
+public class SquareBracketsTest {
+
+    @Test
+    public void testShouldReturnTrueIfSquareBracketsWillBeClosed(){
```
testami należy pokryć wszystkie przypadki wymienione w wymaganiach

BernardBoguszewski commented on src/test/java/pl/sdacademy/javalub11/exercises/FizzBuzzWhizzPrimeNumbersTest.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Test
+    public void testShouldReturnSpecifiedStringsWhenDigitIsPrimeNumberWithTwoOptionsForDigitThreeAndFive() {
```
niedokończony kod nie powinien znajdować się w kodzie pushowanym do code review

exmortuo committed:
ad. Zad 2

Poprawki:
- Część ciała metody isClused wydzielono do osobnej metody prywatnej
- Zmieniono testy na parametryzowane by pokryć wszystkie wymagane
   przypadki.

exmortuo committed:
Poprawki:

- poprawione testy parametryzowane (sekcja when
przeniesiona do metody testującej)

exmortuo committed:
Zad. 4 Wymaganie 1

BernardBoguszewski commented on src/main/java/pl/sdacademy/javalub11/exercises/StringCalculator.java:
```diff
@@ -0,0 +1,20 @@
+public class StringCalculator {
+
+    public Long add(String digitsToAdd) {
+        if(digitsToAdd == null || !digitsToAdd.matches("[0123456789\\,]+") || digitsToAdd.isEmpty()){
```
dla poprawy czytelność można zastosować matchers z //d

exmortuo committed:
Zad. 4 Wymaganie 2

Poprawki:
- w klasie StringCalculator w metodzie Add
zmieniono treść wyrażenia regularnego

exmortuo committed:
Zad. 4 Wymaganie 3

exmortuo committed:
Zad. 4 Wymaganie 4

exmortuo committed:
Zad. 4 Wymaganie 4
- remove unnecessary dependences from pom.xml file
- change delimiter char in test

exmortuo committed:
Dodany plik z opisem zadaniań
